async_asgi_testclient/websocket.py

```diff
@@ -86,11 +87,14 @@ def __aiter__(self):
             (bytes(k.lower(), "utf8"), bytes(v, "utf8"))
             for k, v in self.headers.items()
         ]
+
+        url = URL(self.path)
```

Comment:
Can we use `urllib.parse.urlparse()` instead to avoid `yarl` as a dependency?

```python
import urllib.parse
result = urllib.parse.urlparse("/ws?token=token")
# ...
path = result.path
query_string = result.query.encode("ascii")
```